tests/test_amclient.py

```diff
@@ -436,6 +436,243 @@ def test_close_completed_ingests_no_ingests(self):
         assert isinstance(close_succeeded, list)
         assert len(close_succeeded) == 0
 
+    @vcr.use_cassette('fixtures/vcr_cassettes/pipeline.yaml')
+    def test_get_pipelines(self):
+        """Test getting the pipelines available to the storage service where
+        there is at least one pipeline available to the service.
+        """
+        response = amclient.AMClient(ss_api_key=SS_API_KEY,
+                                     ss_user_name=SS_USER_NAME,
+                                     ss_url=SS_URL).get_pipelines()
+
+        objects = response['objects']
+        pipelines = objects[0]['uuid']
+        resource_uri = objects[0]['resource_uri']
+        assert amclient.is_uuid(pipelines)
+        assert resource_uri == ('/api/v2/pipeline/f914af05-c7d2'
+                                '-4611-b2eb-61cd3426d9d2/')
+        assert isinstance(objects, list)
+        assert len(objects) > 0
+
+    @vcr.use_cassette('fixtures/vcr_cassettes/pipeline_none.yaml')
+    def test_get_pipelines_none(self):
+        """Test getting the pipelines available to the storage service where
+        there is at least one pipeline available to the service.
+        """
+        response = amclient.AMClient(ss_api_key=SS_API_KEY,
+                                     ss_user_name=SS_USER_NAME,
+                                     ss_url=SS_URL).get_pipelines()
+
+        objects = response['objects']
+        assert objects == []
+        assert isinstance(objects, list)
+        assert len(objects) == 0
+
+    @vcr.use_cassette('fixtures/vcr_cassettes/transfer_status.yaml')
+    def test_get_transfer_status(self):
+        """Test the successful return of the status of a transfer for a
+        valid transfer UUID.
+        """
+        response = amclient.AMClient(
+            am_api_key=AM_API_KEY,
+            am_user_name=AM_USER_NAME,
+            am_url=AM_URL,
+            transfer_uuid="63fcc1b0-f83d-47e6"
+                          "-ac9d-a8f8d1fc2ab9").get_transfer_status()
+
+        status = response['status']
+        message = response['message']
+        assert status == 'COMPLETE'
+        assert message == ('Fetched status for 63fcc1b0-f83d-47e6'
+                           '-ac9d-a8f8d1fc2ab9 successfully.')
+
+    @vcr.use_cassette(
+        'fixtures/vcr_cassettes/transfer_status_invalid_uuid.yaml')
+    def test_get_transfer_status_invalid_uuid(self):
+        """Test the successful return of the status for a non-existant
+        transfer in Archivematica.
+        """
+        response = amclient.AMClient(
+            am_api_key=AM_API_KEY,
+            am_user_name=AM_USER_NAME,
+            am_url=AM_URL,
+            transfer_uuid="7bffc8f7-baad-f00d"
```

Comment:
@jrwdunham this pattern where the client holds random attributes isn't very adequate for multi-threaded uses right?